contracts/managerRole.sol

```diff
@@ -0,0 +1,133 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.4;
+
+import "./helpers.sol";
+
+contract managerRole is Helper {
+    constructor(address _instaList, address _instaImplementationM1)
+        Helper(_instaList, _instaImplementationM1)
+    {}
+
+    /**
+     * @dev add a new manager for caller(DSA) along with allowed connectors
+     * @param _manager address to be added as manager
+     * @param _targets array of connector names to be enabled for new manager
+     */
+    function addManagerWithConnectors(
+        address _manager,
+        string[] memory _targets
+    ) public dsaExists(msg.sender) {
+        address[] memory myManagers = dsaManagers[msg.sender];
+        for (uint256 j; j < myManagers.length; j++) {
+            if (myManagers[j] == _manager) {
+                revert("Address already manager");
+            }
+        }
+
+        dsaManagers[msg.sender].push(_manager);
+
+        for (uint256 i; i < _targets.length; i++) {
+            dsaManagerConnectors[msg.sender][_manager].connectorsEnabled[
+                    _targets[i]
+                ] = true;
+
+            dsaManagerConnectors[msg.sender][_manager].allAddedConnectors.push(
+                _targets[i]
+            );
+
+            dsaManagerConnectors[msg.sender][_manager].connectorCount++;
+        }
+    }
+
+    /**
+     * @dev to add connectors to an existing manager of DSA
+     * @param _manager address of manager
+     * @param _targets array connectors to be enabled
+     */
+    function addConnectors(address _manager, string[] memory _targets)
+        public
+        ifManagerExist(msg.sender, _manager)
+        uniqueTargets(_manager, _targets)
+    {
+        for (uint256 i; i < _targets.length; i++) {
+            dsaManagerConnectors[msg.sender][_manager].connectorsEnabled[
+                    _targets[i]
+                ] = true;
+
+            dsaManagerConnectors[msg.sender][_manager].allAddedConnectors.push(
+                _targets[i]
+            );
+
+            dsaManagerConnectors[msg.sender][_manager].connectorCount++;
+        }
+    }
+
+    /**
+     * @dev remove an address from manager role for given DSA
+     * @param _manager address to be removed from manager role
+     */
+    function removeManager(address _manager)
+        public
```

Comment:
valid dsa check

Reply:
The manager won't exist for invalid DSA.

Reply:
Ohh, okay, but we should have DSA check first as well.